google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryTest.java

```diff
@@ -0,0 +1,323 @@
+            .limit(1)
+            .orderBy("a"),
+        "doc5");
+
+    // Test with limits (explicit order by DESC): (a==2) || (b == 1) ORDER BY a LIMIT_TO_LAST 1
+    checkQuerySnapshotContainsDocuments(
+        collection
+            .where(Filter.or(Filter.equalTo("a", 2), Filter.equalTo("b", 1)))
+            .limitToLast(1)
+            .orderBy("a"),
+        "doc2");
+
+    // Test with limits without orderBy (the __name__ ordering is the tie breaker).
+    checkQuerySnapshotContainsDocuments(
+        collection.where(Filter.or(Filter.equalTo("a", 2), Filter.equalTo("b", 1))).limit(1),
+        "doc2");
+  }
+
+  @Test
+  public void orQueryDoesNotIncludeDocumentsWithMissingFields() throws Exception {
+    Map<String, Map<String, Object>> testDocs =
+        map(
+            "doc1", map("a", 1, "b", 0),
+            "doc2", map("b", 1),
+            "doc3", map("a", 3, "b", 2),
+            "doc4", map("a", 1, "b", 3),
+            "doc5", map("a", 1),
+            "doc6", map("a", 2));
+
+    CollectionReference collection = testCollectionWithDocs(testDocs);
+
+    // Query: a==1 || b==1 order by a.
+    // doc2 should not be included because it's missing the field 'a', and we have "orderBy a".
+    Query query1 =
+        collection.where(Filter.or(Filter.equalTo("a", 1), Filter.equalTo("b", 1))).orderBy("a");
+    checkQuerySnapshotContainsDocuments(query1, "doc1", "doc4", "doc5");
+
+    // Query: a==1 || b==1 order by b.
+    // doc5 should not be included because it's missing the field 'b', and we have "orderBy b".
+    Query query2 =
+        collection.where(Filter.or(Filter.equalTo("a", 1), Filter.equalTo("b", 1))).orderBy("b");
+    checkQuerySnapshotContainsDocuments(query2, "doc1", "doc2", "doc4");
+
+    // Query: a>2 || b==1.
+    // This query has an implicit 'order by a'.
+    // doc2 should not be included because it's missing the field 'a'.
+    Query query3 = collection.where(Filter.or(Filter.greaterThan("a", 2), Filter.equalTo("b", 1)));
+    checkQuerySnapshotContainsDocuments(query3, "doc3");
+
+    // Query: a>1 || b==1 order by a order by b.
+    // doc6 should not be included because it's missing the field 'b'.
+    // doc2 should not be included because it's missing the field 'a'.
+    Query query4 =
+        collection
+            .where(Filter.or(Filter.greaterThan("a", 1), Filter.equalTo("b", 1)))
+            .orderBy("a")
+            .orderBy("b");
+    checkQuerySnapshotContainsDocuments(query4, "doc3");
+
+    // Query: a==1 || b==1
+    // There's no explicit nor implicit orderBy. Documents with missing 'a' or missing 'b' should be
+    // allowed if the document matches at least one disjunction term.
+    Query query5 = collection.where(Filter.or(Filter.equalTo("a", 1), Filter.equalTo("b", 1)));
+    checkQuerySnapshotContainsDocuments(query5, "doc1", "doc2", "doc4", "doc5");
+  }
+
+  @Test
+  public void orQueryWithInAndNotIn() throws Exception {
```

Comment:
This test is almost identical to the test on 268, except for 268 is plural. Just double checking that this was intentional.